web-common/src/features/dashboards/dimension-table/dimension-table-utils.ts

```diff
@@ -89,6 +89,25 @@ export function getFilterForComparisonTable(
 // Custom sort that implements the following logic:
 // Input - [ "measure_1", "measure_10", "measure_1_delta", "measure_1_delta_perc", "measure_2", "measure_0", "measure_20" ]
 // Output -  [ "measure_0", "measure_1", "measure_1_delta", "measure_1_delta_perc", "measure_2", "measure_10", "measure_20" ]
+export function customMeasureSortGetter(measureName: string) {
+  return (measure1: string, measure2: string) => {
+    const a_num = parseFloat(
+      measure1
+        .replace(measureName, "")
+        .replace("_delta", ".1")
+        .replace("_delta_perc", ".2")
+    );
+    const b_num = parseFloat(
+      measure2
+        .replace(measureName, "")
+        .replace("_delta", ".1")
+        .replace("_delta_perc", ".2")
+    );
+
+    return a_num - b_num;
```

Comment:
I don't think this works as expected now. Two issues - 

This was based on the fact that measure names will always have a number beside them and the sorting worked as such. But now that we don't have that flexibility, we need to add `someMeasureName_delta` and `someMeasureName_delte_perc` columns adjacent to `someMeasureName`. If the order returned by the `meta` of `topListQuery` can be trusted then we can use that as is and skip the sorting stage.

2. Also this doesn't sort the measures properly even if they are named `measure_0`, `measure_1` etc. There's a test case for `customSortMeasures` in `dimension-table-utils.spec.ts`. You can replace the old method with the new one.

Reply:
I am getting rid of the sorting. Just `splice`ing the comparison columns just after the selected sort measure should work.